internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,405 @@
+│                                                    
+│                                                    
+│            ┌────────────────────────┐                    
+│            │event_filter            │                    
+│           ╱├────────────────────────┤                    
+├──────────○─│ public_id              │                    
+│           ╲│ sink_id                │                    
+│            │ filter_type            |
+|            | filter                 │                    
+│            └────────────────────────┘                                                                                                   
+*/
+
+create table event_type_enm (
+    name text primary key
+        constraint only_predefined_event_types_allowed
+        check (
+            name in (
+                'every',
+                'error',
+                'audit',
+                'observation',
+                'system'
+            )
+        )
+);
+
+comment on table event_type_enm is
+'event_type_enm is an enumeration table for the valid event types within '
+'the domain';
+
+create trigger 
+  immutable_columns
+before
+update on event_type_enm
+  for each row execute procedure immutable_columns('name');
+
+create table event_format_type_enm (
+    name text primary key
+        constraint only_predefined_event_format_types_allowed
+        check (
+            name in (
+                'cloudevents-json',
+                'cloudevents-text',
+                'hclog-text',
+                'hclog-json'
+            )
+        )
+);
+
+comment on table event_format_type_enm is
+'event_format_type_enm is an enumeration table for the valid event format types '
+'within the domain';
+
+create trigger 
+  immutable_columns
+before
+update on event_format_type_enm
+  for each row execute procedure immutable_columns('name');
+
+create table event_filter_type_enm (
+    name text primary key
+        constraint only_predefined_filter_types_allowed
+            check (
+                name in (
+                    'allow',
+                    'deny'
+                )
+            )
+);
+
+comment on table event_filter_type_enm is
+'event_filter_type_enm is an enumeration table for the valid filter types';
+
+create trigger
+  immutable_columns
+before
+update on event_filter_type_enm
+  for each row execute procedure immutable_columns('name');
+
+create table event_config (
+    public_id wt_public_id primary key,
+    scope_id wt_scope_id not null
+        constraint iam_scope_global_fkey
+            references iam_scope_global(scope_id)
+            on delete cascade
+            on update cascade,
+        constraint event_config_scope_id_uq -- only allow one config per scope
+            unique(scope_id),
+    name wt_name,
+    constraint event_config_name_scope_id_uq
+        unique(scope_id, name), 
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version
```

Comment:
Should version use the `update_version_column()` trigger, and if so when should it be incremented? @jimlambrt